### PR TITLE
add mobilenet to gluon model zoo

### PR DESCRIPTION
**Pull request by szha**
@zhreshold

### CODE REVIEW

**Comment by zhreshold** [comment]
Probably this is more readable?
```
self.model = nn.Sequential(
            conv_bn(  3,  32, 2), 
            conv_dw( 32,  64, 1),
            conv_dw( 64, 128, 2),
            conv_dw(128, 128, 1),
            conv_dw(128, 256, 2),
            conv_dw(256, 256, 1),
            conv_dw(256, 512, 2),
            conv_dw(512, 512, 1),
            conv_dw(512, 512, 1),
            conv_dw(512, 512, 1),
            conv_dw(512, 512, 1),
            conv_dw(512, 512, 1),
            conv_dw(512, 1024, 2),
            conv_dw(1024, 1024, 1),
            nn.AvgPool2d(7),
        )
```

File: python/mxnet/gluon/model_zoo/vision/mobilenet.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    classes : int, default 1000
+        Number of classes for the output layer.
+    """
+    def __init__(self, multiplier=1.0, classes=1000, **kwargs):
+        super(MobileNet, self).__init__(**kwargs)
+        with self.name_scope():
+            self.features = nn.HybridSequential(prefix='')
+            with self.features.name_scope():
+                _add_conv(self.features, channels=int(32*multiplier), kernel=3, pad=1, stride=2)
+                _add_conv(self.features, channels=int(32*multiplier), kernel=3, pad=1,
```
Comment (zhreshold) [inline python/mxnet/gluon/model_zoo/vision/mobilenet.py]: if we use depth_wise first in _add_conv_dw, then we can skip the extra _add_conv here and the one at line 70?

File: python/mxnet/gluon/model_zoo/vision/mobilenet.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    net = MobileNet(multiplier, **kwargs)
+    if pretrained:
+        from ..model_store import get_model_file
+        net.load_params(get_model_file('mobilenet'), ctx=ctx)
+    return net
+
+def mobilenet1_0(**kwargs):
+    r"""MobileNet model from the
+    `"MobileNets: Efficient Convolutional Neural Networks for Mobile Vision Applications"
+    <https://arxiv.org/abs/1704.04861>`_ paper, with multiplier 1.0.
```
Comment (piiswrong) [inline python/mxnet/gluon/model_zoo/vision/mobilenet.py]: width multiplier? also add explaination

File: python/mxnet/gluon/model_zoo/vision/mobilenet.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    def __init__(self, multiplier=1.0, classes=1000, **kwargs):
+        super(MobileNet, self).__init__(**kwargs)
+        with self.name_scope():
+            self.features = nn.HybridSequential(prefix='')
+            with self.features.name_scope():
+                _add_conv(self.features, channels=int(32*multiplier), kernel=3, pad=1, stride=2)
+                dw_channels = [int(x*multiplier) for x in [32, 64]+[128]*2+[256]*2+[512]*6+[1024]]
+                channels = [int(x*multiplier) for x in [64]+[128]*2+[256]*2+[512]*6+[1024]*2]
+                stride = [1, 2] * 3 + [1] * 5 + [2, 1]
+                for dw_channels, channels, stride in zip(dw_channels, channels, stride):
```
Comment (piiswrong) [inline python/mxnet/gluon/model_zoo/vision/mobilenet.py]: Use a different name for loop variables

**Pull request merged**